relay-sampling/src/lib.rs

```diff
@@ -449,6 +461,9 @@ impl FieldValueProvider for TraceContext {
 pub struct SamplingConfig {
     /// The sampling rules for the project
     pub rules: Vec<SamplingRule>,
+    /// The id of the next new Rule (used as a generator for unique rule ids)
+    #[serde(default)]
+    pub next_id: Option<u32>,
```

Comment:
Have you considered treating this as something like the "revision" of the config object?